frontend/device/generic/powerd.lua

```diff
@@ -150,12 +150,17 @@ function BasePowerD:setIntensity(intensity)
 end
 
 function BasePowerD:getCapacity()
-    -- NOTE: UIManager *should* be loaded at this point.
-    --       If that doesn't hold, c.f., :stateChanged below.
-    local UIManager = require("ui/uimanager")
-    local now_ts = UIManager:getTime()
+    -- BasePowerD is loaded before UIManager.
+    -- Nothing *currently* calls this before UIManager is actually loaded, but future-proof this anyway.
+    local now_ts
+    if package.loaded["ui/uimanager"] ~= nil then
+        local UIManager = require("ui/uimanager")
+        now_ts = UIManager:getTime()
```

Comment:
(Is this (package > loaded > ui/manager lookups + require() which may do a few more lookups) really less costly (but more verbose) than TimeVal:now() and the C calls ? :)
(If you want to be even more verbose but less costly:

```lua
if self.now_ts_func then
  now_ts = self.now_ts_func()
else if package.loaded["ui/uimanager"] ~= nil then
  local UIManager = require("ui/uimanager")
  self.now_ts_func = function() return UIManager:getTime() end
  now_ts = self.now_ts_func()
else
  now_ts = TimeVal:now()
end
```
)

Reply:
Table lookups are pretty fast, because LuaJIT, but, yeah, caching it might make sense, as we'll only load it once ;).

(i.e., it's just cache the `package.loaded` result and check that, instead of creating a closure).

Reply:
Then again, I'm not entirely convinced it's worth the effort: LuaJIT is *really* good at hash lookups ;).

Reply:
And, yeah, syscalls are much worse, latency-wise (because context switch).

Reply:
Can't you also create a module level `local UIManager` ?
```lua
local UIManager -- will be updated when available
[...]
    local now_ts
    if UIManager then
        now_ts = UIManager:getTime()
    elseif package.loaded["ui/uimanager"] ~= nil then
        UIManager = require("ui/uimanager")
        now_ts = UIManager:getTime()
    else
        now_ts = TimeVal:now()
    end
```
And may be (if PowerD is the only thingy like that), UIManager could signal it to powerd so we don't even need the `elseif package.loaded["ui/uimanager"] ~= nil` then anywhere ?